Framework/DataHandling/src/PulseIndexer.cpp

```diff
@@ -201,8 +201,7 @@ size_t PulseIndexer::getStopEventIndex(const size_t pulseIndex) const {
   const auto pulseIndexEnd = pulseIndex + 1;
 
   // check if the requests have gone past the end - order of if/else matters
-  // const size_t eventIndex = (pulseIndexEnd >= m_roi.back()) ? m_numEvents : m_event_index->operator[](pulseIndexEnd);
-  size_t eventIndex = m_event_index->operator[](pulseIndexEnd);
+  size_t eventIndex = (pulseIndexEnd >= m_event_index->size()) ? m_numEvents : m_event_index->operator[](pulseIndexEnd);
```

Comment:
Just being careful:
Since this _used_ to be `(pulseIndexEnd >= m_roi.back()) ? ...`, I'm wondering why it didn't change to `pulseIndexEnd >= std::min(m_event_index->size(), m_roi.back()) ? std::min(m_roi.back(), m_numEvents) : (*m_event_index)[pulseIndexEnd)`.  Or, is `m_roi.back()` no longer applicable here for some reason?

Reply:
There was a somewhat subtle bug hiding in the old version. When this PR exposed this chunk of code to system tests I had to modify this part. It comes to the fact that number of events only comes into play if the last pulse is being used.